providers/openshift/provider.go

```diff
@@ -568,6 +575,27 @@ func (p *OpenShiftProvider) GetRedeemURL() (*url.URL, error) {
 	return redeemURL, err
 }
 
+func (p *OpenShiftProvider) DeleteSessionToken(session *providers.SessionState) error {
+	if len(session.AccessToken) == 0 {
+		return nil
+	}
+
+	config, err := rest.InClusterConfig()
```

Comment:
This doesn't look right.  Where does `authorizer.SubjectAccessReviewClient` get its kubeconfig from?  You want to store an anonymous kubeconfig right there as a peer.

Reply:
authorizer and authenticator either use supplied kubeconfigs or this - https://github.com/openshift/oauth-proxy/blob/81589f255198ec9aeb1064ec1d20ca0ed822d278/providers/openshift/authorization.go#L71-L100

Reply:
They store their own clients in the config struct conditionally